### PR TITLE
Add minimal calendar support (spec + Phase 1 data model)

### PR DESCRIPTION
**Pull request by kellylford**
## Summary

Implements the approved spec in `docs/planning/calendar-view-pm-dev-spec.md` — minimal calendar support so accepted meeting invites are visible inside QuickMail.

### Resolved design decisions (from review)
- **Shortcut:** `Ctrl+Shift+C` (Calendar). `Ctrl+Shift+L` is already `mail.rules`.
- **Pane layout:** Behaves like the message list; honors `MessageOpenMode` (ReadingPane/Tab/Window) when opening the source invite.
- **Accounts:** All accounts merge into one calendar (unified inbox philosophy).
- **Time zones:** Local time via `ToLocalTime()` for v1.

### What's in this PR (Phase 1 — data model & store)

| File | Change |
|------|--------|
| `CalendarEvent.cs` | Event data class with `DisplayLine` for screen reader announcements |
| `CalendarResponseStatus.cs` | Pending/Accepted/Tentative/Declined enum |
| `ICalendarProvider.cs` | Provider interface (v2 CalDAV/Graph plug-in point) |
| `LocalCacheCalendarProvider.cs` | Harvests events from cached `calendar_ics` text |
| `ICalendarService.cs` / `CalendarService.cs` | Refresh, load, status updates |
| `LocalStoreService.cs` | `CalendarEvent` table (schema v4), `calendar_ics` column on `MessageDetail`, 5 new store methods |
| `ILocalStoreService.cs` | 5 new method signatures |
| `MailMessageDetail.cs` | `CalendarIcs` property (raw ICS text, persisted) |
| `ImapMailService.cs` | Populates `CalendarIcs` when fetching `text/calendar` parts |
| `StubServices.cs` + 5 test stores | Implement new interface methods |
| `CalendarStoreTests.cs` | 7 tests: upsert/load round-trip, conflict preserves status, status update, delete, sort, migration coexistence, ICS harvest |

### What's NOT in this PR (Phases 2–4, to follow on this branch)
- **Phase 2:** `CalendarViewModel`, `MainViewModel` integration, `view.calendar` command, config settings
- **Phase 3:** `CalendarPane.xaml` UserControl, F6 ring integration, keyboard wiring
- **Phase 4:** Sync-triggered harvest, response-status update on accept/decline, `--online` empty state

### Test results
All 7 Phase 1 tests pass. Existing tests compile and pass (stub updates were required across 6 test files that implement `ILocalStoreService`).

### Out of scope (v1)
CalDAV/Graph sync, event creation/editing, reminders, `RRULE` expansion, month/week grid, all-day band, per-account filter, time-zone selector.

```
dotnet test QuickMail.Tests/QuickMail.Tests.csproj -c Debug --filter CalendarStoreTests
# Passed: 7, Failed: 0
```

### CODE REVIEW

File: QuickMail/Services/LocalStoreService.cs [hunk 1]
```diff
@@ -713,4 +752,128 @@ private sealed record AttachmentMeta(
...
+
+    public async Task<List<(Guid AccountId, string FolderName, string MessageId, string IcsText)>> LoadAllCalendarIcsAsync()
+    {
+        var list = new List<(Guid, string, string, string)>();
+        await using var conn = await OpenAsync();
+        await using var cmd = conn.CreateCommand();
+        cmd.CommandText = """
+            SELECT account_id, folder_name, unique_id, calendar_ics
+            FROM MessageDetail
+            WHERE calendar_ics IS NOT NULL AND calendar_ics != '';
```
Comment (kellylford) [inline QuickMail/Services/LocalStoreService.cs:866]: **🔴 Bug #1 — No `ORDER BY` means a reschedule can silently revert to old data**

When an organizer reschedules a meeting, both the original (SEQUENCE:0) and updated (SEQUENCE:1) `MessageDetail` rows exist with the same ICS UID. `LoadAllCalendarIcsAsync` has no `ORDER BY`, so SQLite returns them in heap order. `HarvestAsync` upserts unconditionally (the `ON CONFLICT` clause updates `summary`, `start_time_ticks`, `location`, etc.). If SEQUENCE:1 is processed first, the old SEQUENCE:0 row overwrites the new time and title on the next pass.

**Fix:** Add an `ORDER BY` on message date so newer rows win:
```sql
SELECT account_id, folder_name, unique_id, calendar_ics
FROM MessageDetail d
JOIN MessageSummary s USING (unique_id, account_id, folder_name)
WHERE d.calendar_ics IS NOT NULL AND d.calendar_ics != ''
ORDER BY s.date_ticks ASC
```
(Process oldest first so newest wins the final upsert.) Alternatively, add sequence-number comparison inside `HarvestAsync` and skip if the stored sequence is already higher.

File: QuickMail/ViewModels/MainViewModel.cs [hunk 2]
```diff
@@ -4819,11 +4928,94 @@ private async Task SendIcsReply(string partStat, string actionLabel)
...
+    /// Runs on the UI thread via Dispatcher so the CalendarService refresh is safe.
+    /// </summary>
+    private void ScheduleCalendarHarvest()
+    {
+        if (_calendarService == null || CalendarVm == null || OnlineMode) return;
+
+        // Reset the timer — if a previous harvest was pending, it gets pushed back 2s.
+        _calendarHarvestTimer ??= new System.Threading.Timer(_ =>
+        {
+            System.Windows.Application.Current?.Dispatcher.InvokeAsync(async () =>
```
Comment (kellylford) [inline QuickMail/ViewModels/MainViewModel.cs:5009]: **🔴 Bug #2 — `Dispatcher.InvokeAsync(async lambda)` silently swallows all harvest exceptions**

The `async () => { await _calendarService.RefreshAsync(); }` lambda matches `InvokeAsync<Task>(Func<Task>)`, returning `DispatcherOperation<Task>`. The outer operation is discarded; the inner `Task` is never awaited or observed. Any exception from `RefreshAsync()` (SQLite locked, I/O error, parse failure) is silently lost — no log, no user announcement, calendar stays stale.

**Fix:** Use a dedicated `async void` helper with `try/catch`:
```csharp
_calendarHarvestTimer ??= new System.Threading.Timer(_ =>
    Application.Current?.Dispatcher.InvokeAsync(DoHarvestAsync),
    null, Timeout.Infinite, Timeout.Infinite);

private async void DoHarvestAsync()
{
    try
    {
        if (_calendarService == null || CalendarVm == null) return;
        await _calendarService.RefreshAsync();
        if (IsCalendarView) CalendarVm.ApplyFiltersFromExternalUpdate();
    }
    catch (Exception ex) { LogService.Debug($"[CALENDAR] harvest failed: {ex.Message}"); }
}
```

File: QuickMail/ViewModels/CalendarViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+/// ViewModel for the calendar pane. Owns the event list, today filter, refresh,
+/// and open-source-message command. Announcements are surfaced via
+/// <see cref="AnnouncementRequested"/>; the View subscribes and calls
+/// <c>AccessibilityHelper.Announce</c>.
+/// </summary>
+public partial class CalendarViewModel : ObservableObject
+{
+    private readonly ICalendarService _calendarService;
+    private readonly bool _onlineMode;
+    private readonly bool _showDeclinedEvents;
```
Comment (kellylford) [inline QuickMail/ViewModels/CalendarViewModel.cs:24]: **🔴 Bug #3 — `_showDeclinedEvents` is frozen at construction; toggling the setting has no effect until app restart**

`CalendarVm` is created once in `MainViewModel`'s constructor and never replaced. If the user toggles "Show declined events" in Settings and saves, `_showDeclinedEvents` keeps its original value and `ApplyFilters()` continues hiding (or showing) declined events as if the setting never changed.

**Fix:** Either read the setting on each `ApplyFilters()` call:
```csharp
var showDeclined = _configService.Load().ShowDeclinedEvents;
```
…or expose a `void SetShowDeclinedEvents(bool value)` method and have `MainViewModel` call it after saving config.

File: QuickMail/Models/CalendarEvent.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    /// <summary>ICS METHOD (REQUEST, CANCEL, REPLY, etc.).</summary>
+    public string? Method { get; set; }
+
+    /// <summary>MessageId (per-folder storage key) of the source invite email.</summary>
+    public string SourceMessageId { get; set; } = string.Empty;
+
+    /// <summary>Folder name of the source invite email.</summary>
+    public string SourceFolder { get; set; } = string.Empty;
+
+    [ObservableProperty]
```
Comment (kellylford) [inline QuickMail/Models/CalendarEvent.cs:44]: **🟠 Bug #4 — `DisplayLine` is not notified when `ResponseStatus` changes — stale screen reader name**

`[ObservableProperty]` on `_responseStatus` only fires `PropertyChanged("ResponseStatus")`. `DisplayLine` reads `ResponseStatus` (line 85) but is a plain computed property — no notification is generated for it. `MainWindow.xaml` binds `AutomationProperties.Name` of each `ListBoxItem` to `DisplayLine`, so after accept/decline the screen reader keeps announcing the old status.

**Fix:**
```csharp
[ObservableProperty]
[NotifyPropertyChangedFor(nameof(DisplayLine))]
private CalendarResponseStatus _responseStatus = CalendarResponseStatus.Pending;
```

**Review by kellylford — COMMENTED** [review]
## Code Review — Calendar branch (Phase 1)

15 findings across correctness, CLAUDE.md compliance, architecture, and efficiency. Inline comments are on the specific lines; this table is the full summary.

| # | Severity | File | Issue |
|---|---|---|---|
| 1 | 🔴 Bug | `LocalStoreService.cs` / `LocalCacheCalendarProvider.cs` | No `ORDER BY` on ICS scan — reschedule overwrites updated event with stale data |
| 2 | 🔴 Bug | `MainViewModel.cs` | `Dispatcher.InvokeAsync(async lambda)` — harvest exceptions silently swallowed |
| 3 | 🔴 Bug | `CalendarViewModel.cs` | `_showDeclinedEvents` frozen at construction — settings change silently ignored |
| 4 | 🟠 Bug | `CalendarEvent.cs` | `DisplayLine` not notified when `ResponseStatus` changes — stale screen reader name |
| 5 | 🟠 Bug | `LocalStoreService.cs` | `DateTimeOffset.MinValue` renders as year-0001 when source invite is purged |
| 6 | 🟠 Bug | `MainWindow.xaml.cs` | No modifier guard on `Key.T` / `Key.Enter` — `Ctrl+T`, `Alt+Enter` fire wrong commands |
| 7 | 🟠 CLAUDE.md | `MainWindow.xaml.cs` | `T` and `Enter` hardcoded, not registered in `CommandRegistry` |
| 8 | 🟠 CLAUDE.md | `MainWindow.xaml` | `AutomationProperties.Name` bound to full sentence — violates CLAUDE.md + causes double-reading |
| 9 | 🟡 Architecture | `CalendarService.cs` | `is LocalCacheCalendarProvider` cast breaks `ICalendarProvider` abstraction |
| 10 | 🟡 Resource leak | `MainViewModel.cs` | `_calendarHarvestTimer` never stopped or disposed (CA1001) |
| 11 | 🟡 Bug | `MainViewModel.cs` | In-flight folder fetch not cancelled when switching to Calendar |
| 12 | 🟡 CLAUDE.md | `MainWindow.xaml.cs` | Domain object (`MailMessageSummary` stub) constructed in code-behind |
| 13 | 🟡 Convention | `MainWindow.xaml.cs` | Calendar event subscriptions never unsubscribed in `OnClosed` |
| 14 | 🟡 Test gap | `CalendarViewModel.cs` | Tests assert `VisibleEvents`; UI binds `Events` — different objects |
| 15 | 🔵 Efficiency | `LocalCacheCalendarProvider.cs` | Per-row DB connection in harvest loop — slow on large mailboxes |

File: QuickMail/Services/LocalStoreService.cs [hunk 5]
```diff
@@ -602,11 +639,13 @@ JOIN MessageSummary s USING (unique_id, account_id, folder_name)
             ReplyTo       = r.GetString(2),
             PlainTextBody = r.GetString(3),
             HtmlBody      = r.GetString(4),
-            From          = r.GetString(5),
-            Subject       = r.GetString(6),
-            Date          = new DateTimeOffset(r.GetInt64(7), TimeSpan.Zero),
-            IsRead        = r.GetInt64(8) != 0,
+            From          = r.IsDBNull(5) ? string.Empty : r.GetString(5),
+            Subject       = r.IsDBNull(6) ? "(no subject)" : r.GetString(6),
+            Date          = r.IsDBNull(7) ? DateTimeOffset.MinValue : new DateTimeOffset(r.GetInt64(7), TimeSpan.Zero),
```
Comment (kellylford) [inline QuickMail/Services/LocalStoreService.cs:644]: **🟠 Bug #5 — `DateTimeOffset.MinValue` renders as "January 01, 0001" in the reading pane**

The LEFT JOIN correctly handles a purged `MessageSummary` row, but sets `Date = DateTimeOffset.MinValue` when column 7 is NULL. `MainWindow.xaml` line 926 binds this to `StringFormat='{}{0:f}'` with no MinValue guard, so the Date header displays "Monday, January 01, 0001 12:00 AM" when the user opens a calendar event whose invite has scrolled out of the sync window.

**Fix:** Use a nullable fallback:
```csharp
Date = r.IsDBNull(7) ? DateTimeOffset.Now : new DateTimeOffset(r.GetInt64(7), TimeSpan.Zero),
```
…or change the field type to `DateTimeOffset?` and handle null in the XAML `FallbackValue`.

File: QuickMail/Views/MainWindow.xaml.cs [hunk 6]
```diff
@@ -1462,6 +1479,102 @@ private void FocusMessageListFirstItem()
...
+            && Keyboard.Modifiers == ModifierKeys.None
+            && CalendarList.Items.Count == 0)
+        {
+            e.Handled = true;
+            return;
+        }
+
+        switch (e.Key)
+        {
+            case Key.Enter:
```
Comment (kellylford) [inline QuickMail/Views/MainWindow.xaml.cs:1558]: **🟠 Bug #6 — No modifier guard on `Key.Enter` and `Key.T`; modifier chords fire unintended commands**

`Ctrl+T` (a common new-tab convention), `Alt+T`, `Shift+Enter`, and `Alt+Enter` all fall through to `ToggleTodayFilter` or `OpenSourceMessage` when the calendar list has focus. The empty-list arrow guard just above (line 1548) correctly checks `Keyboard.Modifiers == ModifierKeys.None` — the same guard is missing here.

**Fix:**
```csharp
case Key.Enter when Keyboard.Modifiers == ModifierKeys.None:
    ...
case Key.T when Keyboard.Modifiers == ModifierKeys.None:
    ...
```

File: QuickMail/Views/MainWindow.xaml.cs [hunk 7]
```diff
@@ -1462,6 +1479,102 @@ private void FocusMessageListFirstItem()
...
+                if (vm.SelectedEvent != null)
+                {
+                    vm.OpenSourceMessageCommand.Execute(vm.SelectedEvent);
+                    e.Handled = true;
+                }
+                break;
+
+            case Key.T:
```
Comment (kellylford) [inline QuickMail/Views/MainWindow.xaml.cs:1566]: **🟠 CLAUDE.md #7 — `T` is hardcoded here, not registered in `CommandRegistry`**

CLAUDE.md: *"Every user-facing keyboard shortcut must be registered in `CommandRegistry` via `Register(new CommandDefinition(...))`. Register first, hardcode never."* The two permitted hardcoded exceptions are `Ctrl+Shift+P` and the `Ctrl+0–3/9/Y/F6` pane-jump family. Plain `T` (toggle today filter) and `Enter` (open source message) are neither.

Neither `calendar.toggleTodayFilter` nor `calendar.openSourceMessage` appear anywhere in `RegisterCommands`. These shortcuts don't appear in the keyboard customizations dialog and can't be rebound.

**Fix:** Register both commands in `RegisterCommands`:
```csharp
registry.Register(new CommandDefinition(
    id: "calendar.toggleTodayFilter", category: "View", title: "Toggle today filter",
    execute: () => CalendarVm?.ToggleTodayFilterCommand.Execute(null),
    defaultKey: Key.T,
    isAvailable: () => CalendarVm != null && IsCalendarView));

registry.Register(new CommandDefinition(
    id: "calendar.openSourceMessage", category: "Mail", title: "Open calendar invite",
    execute: () => CalendarVm?.OpenSourceMessageCommand.Execute(CalendarVm.SelectedEvent),
    defaultKey: Key.Enter,
    isAvailable: () => CalendarVm != null && IsCalendarView && CalendarVm.SelectedEvent != null));
```
Then dispatch through the registry in `CalendarList_PreviewKeyDown` instead of hardcoding.

File: QuickMail/Views/MainWindow.xaml [hunk 8]
```diff
@@ -1039,10 +1040,44 @@
...
+                             KeyboardNavigation.DirectionalNavigation="Contained"
+                             VirtualizingPanel.IsVirtualizing="True"
+                             VirtualizingPanel.VirtualizationMode="Recycling"
+                             ScrollViewer.HorizontalScrollBarVisibility="Disabled"
+                             ScrollViewer.VerticalScrollBarVisibility="Auto"
+                             GotKeyboardFocus="CalendarList_GotKeyboardFocus"
+                             PreviewKeyDown="CalendarList_PreviewKeyDown">
+                        <ListBox.ItemContainerStyle>
+                            <Style TargetType="ListBoxItem" BasedOn="{StaticResource {x:Type ListBoxItem}}">
+                                <Setter Property="AutomationProperties.Name" Value="{Binding DisplayLine}"/>
```
Comment (kellylford) [inline QuickMail/Views/MainWindow.xaml:1064]: **🟠 CLAUDE.md #8 — `AutomationProperties.Name` is bound to a full sentence; violates CLAUDE.md and causes double-reading**

CLAUDE.md: *"AutomationProperties.Name is a short label only. No descriptions, no role names, no keyboard shortcuts, no sentences."* `DisplayLine` produces `"Team standup, today 10:00 to 10:30, Accepted. Location: Zoom."` — that's a sentence.

Additionally, the `DataTemplate` (line 1072) also has `<TextBlock Text="{Binding DisplayLine}"/>` as an inner element. Screen readers announce the container's automation name, then enumerate both inner TextBlocks — `DisplayLine` is spoken twice (plus `Summary` once).

**Fix:**
1. Use `Summary` as the short `AutomationProperties.Name`.
2. Surface the full sentence via `AutomationProperties.ItemStatus` (which screen readers announce as supplementary context, not as the element's identity).
3. Add `AutomationProperties.AccessibilityView="Raw"` to the inner `DisplayLine` TextBlock to suppress duplicate reading.

```xml
<Setter Property="AutomationProperties.Name" Value="{Binding Summary}"/>
<Setter Property="AutomationProperties.ItemStatus" Value="{Binding DisplayLine}"/>
```

File: QuickMail/Services/CalendarService.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        get
+        {
+            lock (_lock) return _events.ToList();
+        }
+    }
+
+    public async Task RefreshAsync(CancellationToken ct = default)
+    {
+        // Harvest from the local cache (no-op for non-harvesting providers).
+        if (_provider is LocalCacheCalendarProvider local)
```
Comment (kellylford) [inline QuickMail/Services/CalendarService.cs:37]: **🟡 Architecture #9 — Concrete type check breaks the `ICalendarProvider` abstraction**

`HarvestAsync` is not on the interface, so this cast is the only way to call it — but it means any future provider (CalDAV, Graph) silently skips harvesting. `RefreshAsync` falls through to `LoadEventsAsync` only, returning stale data. The spec explicitly calls out CalDAV/Graph as v2 targets.

**Fix:** Add `HarvestAsync` to `ICalendarProvider` with a no-op default:
```csharp
/// <summary>No-op for providers that don't harvest (e.g. CalDAV push). Override to scan and import.</summary>
Task HarvestAsync(CancellationToken ct = default) => Task.CompletedTask;
```
Then in `RefreshAsync`:
```csharp
await _provider.HarvestAsync(ct);
var loaded = await _provider.LoadEventsAsync(ct);
```

File: QuickMail/ViewModels/MainViewModel.cs [hunk 10]
```diff
@@ -4819,11 +4928,94 @@ private async Task SendIcsReply(string partStat, string actionLabel)
...
+    /// <summary>
+    /// Schedules a debounced calendar harvest 2 seconds after the last FolderSynced event.
+    /// Runs on the UI thread via Dispatcher so the CalendarService refresh is safe.
+    /// </summary>
+    private void ScheduleCalendarHarvest()
+    {
+        if (_calendarService == null || CalendarVm == null || OnlineMode) return;
+
+        // Reset the timer — if a previous harvest was pending, it gets pushed back 2s.
+        _calendarHarvestTimer ??= new System.Threading.Timer(_ =>
```
Comment (kellylford) [inline QuickMail/ViewModels/MainViewModel.cs:5007]: **🟡 Resource leak #10 — `_calendarHarvestTimer` is never stopped or disposed (CA1001)**

CLAUDE.md IDisposable rules: *"For every new `IDisposable` implementation, identify where `Dispose()` is called before the PR is merged."* `System.Threading.Timer` implements `IDisposable`. `MainViewModel` has no `Dispose()` method and `App.OnExit` doesn't touch this field. A pending timer callback can fire during or after app shutdown.

**Fix:** Either implement `IDisposable` on `MainViewModel` (add `_calendarHarvestTimer?.Dispose()` to `Dispose()` and call it from `App.OnExit`), or suppress CA1001 with a justified comment explaining why disposal is a no-op for this process-lifetime singleton.

File: QuickMail/ViewModels/MainViewModel.cs [hunk 11]
```diff
@@ -2392,6 +2454,35 @@ private async Task SelectFolderAsync(MailFolderModel? folder)
         }
     }
 
+    /// <summary>
+    /// Activates the calendar view: clears message-list state, loads calendar events,
+    /// and requests focus to the event list. Called when the user selects the
+    /// Calendar virtual folder from the folder tree.
+    /// </summary>
+    private async Task SelectCalendarAsync()
```
Comment (kellylford) [inline QuickMail/ViewModels/MainViewModel.cs:2462]: **🟡 Bug #11 — In-flight folder fetch not cancelled when switching to Calendar**

`SelectCalendarAsync` calls `SetMessages([])` to clear the list but does not cancel `_folderCts`. Compare with every other folder selection path (lines 960–961, 2789–2790, etc.), which all call `_folderCts?.Cancel(); ReplaceCts(ref _folderCts, out var ct);` before doing any work. Without this, an in-progress `FetchFolderAsync` can complete after the calendar is active, re-call `SetMessages(...)`, and trigger a spurious `FocusActiveMessagePanel` dispatch.

**Fix:** Add the same cancellation pattern at the start of `SelectCalendarAsync`:
```csharp
_folderCts?.Cancel();
ReplaceCts(ref _folderCts, out _);
```

File: QuickMail/Views/MainWindow.xaml.cs [hunk 12]
```diff
@@ -2101,10 +2214,44 @@ private void CloseReadingPane()
...
+    {
+        LogService.Debug($"[CALENDAR] OpenSourceMessageFromCalendar accountId={accountId} folder={folder} messageId={messageId}");
+
+        // Find the account and select it so SelectMessageAsync has the right context.
+        var account = _vm.Accounts.FirstOrDefault(a => a.Id == accountId);
+        if (account != null)
+            _vm.SelectedAccount = account;
+
+        // Construct a summary stub; SelectMessageAsync will fetch the full detail.
+        var summary = new MailMessageSummary
```
Comment (kellylford) [inline QuickMail/Views/MainWindow.xaml.cs:2232]: **🟡 CLAUDE.md #12 — Domain object constructed in code-behind**

CLAUDE.md: *"Not permitted in .xaml.cs: business logic, data transformation."* Translating `(accountId, folder, messageId)` from the calendar event into a `MailMessageSummary` stub is data transformation — if `MailMessageSummary` ever gains a required routing property, the stub silently defaults it wrong.

**Fix:** Move the stub construction into `MainViewModel` — add a method `OpenCalendarSourceMessage(Guid accountId, string folder, string messageId)` that builds the stub internally. Code-behind subscribes to `OpenSourceMessageRequested` and calls `_vm.OpenCalendarSourceMessageCommand.Execute((accountId, folder, messageId))`, keeping all data transformation in the VM.

File: QuickMail/Views/MainWindow.xaml.cs [hunk 13]
```diff
@@ -257,6 +257,17 @@ public MainWindow(
 
         vm.TabPromoteToWindowRequested += PromoteTabToWindow;
 
+        // Calendar list subscriptions
+        vm.CalendarPaneFocusRequested += () => Dispatcher.InvokeAsync(
+            () => FocusCalendarList(), System.Windows.Threading.DispatcherPriority.Input);
+        if (vm.CalendarVm != null)
+        {
+            vm.CalendarVm.AnnouncementRequested += (text, category) =>
```
Comment (kellylford) [inline QuickMail/Views/MainWindow.xaml.cs:265]: **🟡 Convention #13 — Calendar event subscriptions are never unsubscribed in `OnClosed`**

Three subscriptions are wired here:
- `vm.CalendarPaneFocusRequested +=`
- `vm.CalendarVm.AnnouncementRequested +=`
- `vm.CalendarVm.OpenSourceMessageRequested +=`

None are removed in `OnClosed`. CLAUDE.md: *"Event subscriptions on dialog VMs must be cleaned up."* This is inert in the current single-window architecture (all objects go out of scope together), but should be cleaned up before any path exists that creates a second `MainWindow` against the same VM.

**Fix:** Unsubscribe in `OnClosed`:
```csharp
if (vm.CalendarVm != null)
{
    vm.CalendarVm.AnnouncementRequested -= _calendarAnnouncementHandler;
    vm.CalendarVm.OpenSourceMessageRequested -= _calendarOpenSourceHandler;
}
vm.CalendarPaneFocusRequested -= _calendarFocusHandler;
```

File: QuickMail/ViewModels/CalendarViewModel.cs [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+    /// The View subscribes, constructs a <see cref="MailMessageSummary"/> from the args,
+    /// and routes through the existing <c>SelectMessageCommand</c> so
+    /// <see cref="MessageOpenMode"/> is honored.
+    /// </summary>
+    public event Action<Guid, string, string>? OpenSourceMessageRequested;
+
+    [ObservableProperty]
+    private BatchObservableCollection<CalendarEvent> _events = [];
+
+    /// <summary>All events after filtering (today filter + declined filter), sorted by start time.</summary>
```
Comment (kellylford) [inline QuickMail/ViewModels/CalendarViewModel.cs:43]: **🟡 Test gap #14 — Tests assert `VisibleEvents`; XAML binds `Events` — different objects**

`VisibleEvents` returns `_filteredEvents` (a `List<CalendarEvent>`). The XAML binds to `CalendarVm.Events` (a `BatchObservableCollection<CalendarEvent>`). Both are populated from `ApplyFilters()` — so they're always in sync — but a bug that corrupts `Events` without touching `_filteredEvents` (e.g. a wrong `BeginBatchScope` call, missing `Clear()`, an off-by-one in the batch) would be invisible to all current tests.

**Fix:** Either expose `Events` as the test-facing surface (assert on `vm.Events.Count`) or add at least one test that asserts on `vm.Events` directly to close the coverage gap.

File: QuickMail/Services/LocalCacheCalendarProvider.cs [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+                OrganizerName    = model.OrganizerName ?? string.Empty,
+                StartTimeTicks   = model.StartTime?.ToUniversalTime().Ticks,
+                EndTimeTicks     = model.EndTime?.ToUniversalTime().Ticks,
+                Sequence         = model.Sequence,
+                Method           = model.Method,
+                SourceMessageId  = messageId,
+                SourceFolder     = folder,
+                ResponseStatus   = isCancel ? CalendarResponseStatus.Cancelled : CalendarResponseStatus.Pending,
+            };
+            await _store.UpsertCalendarEventAsync(evt);
```
Comment (kellylford) [inline QuickMail/Services/LocalCacheCalendarProvider.cs:81]: **🔵 Efficiency #15 — Per-row DB connection in harvest loop; slow on large mailboxes**

Each `UpsertCalendarEventAsync` call opens a new `SqliteConnection`, begins a transaction, executes one insert, and commits. For a user with hundreds of invite emails, the initial harvest can take several seconds — and this call is on the `await`ed `CalendarVm.LoadAsync()` path, so the calendar pane appears to hang on first open.

**Fix:** Batch the upserts inside a single connection and transaction:
```csharp
var events = new List<CalendarEvent>();
foreach (var (accountId, folder, messageId, icsText) in rows)
{
    // ... parse and build evt ...
    events.Add(evt);
}
await _store.UpsertCalendarEventBatchAsync(events);
```
Or accept a `SqliteConnection` parameter on `UpsertCalendarEventAsync` so the harvest can share one open connection.

**Comment by kellylford** [comment]
## Follow-up review — Address-all-15 commit

I read the current branch state against all 15 original findings. Here's the scorecard, then the three remaining items.

---

### ✅ All 15 original findings — verified fixed

| # | Finding | Fix verified |
|---|---------|-------------|
| 1 | No `ORDER BY` on ICS scan | `ORDER BY s.date_ticks ASC` added; JOIN added to get the date |
| 2 | `Dispatcher.InvokeAsync(async lambda)` swallows exceptions | Replaced with `async void DoCalendarHarvest()` + try/catch |
| 3 | `_showDeclinedEvents` frozen at construction | `SetShowDeclinedEvents(bool)` added; called from `ApplySettings` after settings dialog closes |
| 4 | `DisplayLine` not notified on `ResponseStatus` change | `[NotifyPropertyChangedFor(nameof(DisplayLine))]` added |
| 5 | `DateTimeOffset.MinValue` renders as year-0001 | `DataTrigger` in XAML hides the date field when `Date == DateTimeOffset.MinValue` |
| 6 | No modifier guard on `Key.T` / `Key.Enter` | `if (Keyboard.Modifiers != ModifierKeys.None) return;` added at top of handler |
| 7 | `T` and `Enter` hardcoded, not in `CommandRegistry` | `calendar.toggleTodayFilter` and `calendar.openSourceMessage` registered; handler dispatches through `_registry.FindByGesture()` |
| 8 | `AutomationProperties.Name` bound to full sentence | Name is now `{Binding Summary}` (short label); `DisplayLine` moved to `ItemStatus`; inner TextBlock suppressed with `AutomationProperties.Name=""` |
| 9 | `is LocalCacheCalendarProvider` cast breaks abstraction | `HarvestAsync` added to `ICalendarProvider` with default no-op; `CalendarService.RefreshAsync` calls `_provider.HarvestAsync()` |
| 10 | `_calendarHarvestTimer` never disposed | `internal void Dispose()` added to `MainViewModel`; called from `App.OnExit` |
| 11 | In-flight folder fetch not cancelled on calendar switch | `_folderCts?.Cancel(); ReplaceCts(ref _folderCts, out _);` added to `SelectCalendarAsync()` |
| 12 | `MailMessageSummary` stub constructed in code-behind | Code-behind calls `_vm.OpenCalendarSourceMessageCommand`; stub construction moved into `MainViewModel` |
| 13 | Event subscriptions never unsubscribed | All three calendar subscriptions removed in `OnClosed` |
| 14 | Tests assert `VisibleEvents`; XAML binds `Events` | **Partially — see below** |
| 15 | Per-row DB connection in harvest loop | `UpsertCalendarEventBatchAsync` added; harvest collects all events then calls the batch method |

---

### Remaining items (3)

#### 🟡 #14 — Test gap still present

`VisibleEvents` still returns `_filteredEvents` (a plain `List<CalendarEvent>`), while the XAML binds `CalendarVm.Events` (the `BatchObservableCollection<CalendarEvent>`). Both are populated in `ApplyFilters()` so they're always in sync at runtime, but a bug in the `BatchObservableCollection` batch scope — a missing `Clear()`, a double-add, an off-by-one inside `BeginBatchScope` — would be invisible to every test.

The fix is small: eliminate `_filteredEvents` entirely and make `VisibleEvents` return `Events`.

```csharp
// CalendarViewModel.cs — replace
public IReadOnlyList<CalendarEvent> VisibleEvents => _filteredEvents;
private List<CalendarEvent> _filteredEvents = [];
// with
public IReadOnlyList<CalendarEvent> VisibleEvents => Events;
```

In `ApplyFilters()`, replace the `_filteredEvents` assignment with a local:

```csharp
private void ApplyFilters()
{
    var all = _calendarService.Events;
    IEnumerable<CalendarEvent> filtered = all;
    if (!_showDeclinedEvents)
        filtered = filtered.Where(e => e.ResponseStatus != CalendarResponseStatus.Declined);
    filtered = filtered.Where(e => e.ResponseStatus != CalendarResponseStatus.Cancelled);
    if (IsTodayFilter)
    {
        var today = DateTime.Today;
        filtered = filtered.Where(e => e.StartTime.HasValue && e.StartTime.Value.Date == today);
    }
    var sorted = filtered.OrderBy(e => e.StartTimeTicks ?? long.MaxValue).ToList();
    using (Events.BeginBatchScope())
    {
        Events.Clear();
        foreach (var evt in sorted)
            Events.Add(evt);
    }
}
```

Now `VisibleEvents` IS `Events` — the tests automatically exercise the right collection.

---

#### 🔵 New regression from the #1 fix — `LoadAllCalendarIcsAsync` INNER JOIN silently drops purged invites

The `ORDER BY date_ticks` fix correctly added a JOIN to `MessageSummary`. The join is INNER, so any `MessageDetail` row whose `MessageSummary` has been purged from the sync window is silently excluded from future harvests. This is inconsistent with the LEFT JOIN added to `LoadDetailAsync` for exactly this scenario (opening a calendar event whose invite has scrolled out of the sync window).

In practice the impact is narrow: a purged invite's event was already upserted to `CalendarEvent` on the first harvest, so the event stays in the calendar correctly. A re-harvest just won't re-process that row. The only scenario where this matters is a reschedule whose source message summary was purged before the re-harvest ran — an unusual combination.

Consider a LEFT JOIN with a fallback date:

```sql
SELECT d.account_id, d.folder_name, d.unique_id, d.calendar_ics
FROM MessageDetail d
LEFT JOIN MessageSummary s ON s.unique_id   = d.unique_id
                           AND s.account_id  = d.account_id
                           AND s.folder_name = d.folder_name
WHERE d.calendar_ics IS NOT NULL AND d.calendar_ics != ''
ORDER BY COALESCE(s.date_ticks, 0) ASC;
```

Rows without a summary get `date_ticks = 0` and sort first, so they're overwritten by any newer message with a real date — which is the same correctness guarantee the fix was targeting.

---

#### 🔵 `RefreshCommand` test uses `Task.Delay(100)` — timing-dependent

```csharp
vm.RefreshCommand.Execute(null);
await Task.Delay(100);  // fragile
```

`AsyncRelayCommand` exposes `ExecuteAsync`, which is directly awaitable:

```csharp
await vm.RefreshCommand.ExecuteAsync(null);
// no Task.Delay needed
```

This removes the 100 ms race and makes the test deterministic.

---

### Verdict

All 15 original findings are addressed. The three items above are the only remaining gaps:

- **#14** (test gap — wrong collection tested) is the highest priority; it's a 5-line fix and makes the tests actually exercise the XAML-bound collection.
- The INNER JOIN regression is a low-severity edge case but worth a one-line fix for consistency with `LoadDetailAsync`.
- The `Task.Delay(100)` is a straightforward test cleanup.

None of these are runtime bugs in the happy path, but I'd land at least #14 before merging since it's what the original review asked for.
